### PR TITLE
compile, deploy, and run Dartino app on STM32 discovery board

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew Initial support for Dartino, but more cleanup needed

### CODE REVIEW

File: lib/sdk/dartino_sdk.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    if (isLinux) {
+      //TODO(danrubel) need to test this on Linux
+
+    }
+    if (isWindows) {
+      // TODO
+    }
+    atom.notifications.addError('OS not supported yet.', dismissable: true);
+    return false;
+    return false;
```
Comment (devoncarew) [inline lib/sdk/dartino_sdk.dart:83]: Several returns :)

File: lib/sdk/dartino_sdk.dart [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  @override
+  Future<bool> verifyInstall() async {
+    if (!checkSdkFile(
+        'Dartino',
+        sdkRootPath,
+        ['platforms/stm32f746g-discovery/bin/build.sh'],
+        'Please download the SDK set the Dartino SDK path in\n'
+        'Settings > Packages > $pluginId > Dartino root directory'))
+      return false;
```
Comment (devoncarew) [inline lib/sdk/dartino_sdk.dart:94]: braces?

**Comment by devoncarew** [comment]
lgtm!

Do you have travis set up for this repo (or intend to use it)? If not, I'm happy to help.

File: lib/sdk/dartino_sdk.dart (hunk 1 above)
Comment (danrubel) [inline lib/sdk/dartino_sdk.dart:83]: Good point. Addressed in https://github.com/dartino/atom-dartino/pull/3

File: lib/sdk/dartino_sdk.dart (hunk 2 above)
Comment (danrubel) [inline lib/sdk/dartino_sdk.dart:94]: Good point. Addressed in https://github.com/dartino/atom-dartino/pull/3

**Comment by danrubel** [comment]
I want travis setup for this repo, but it's not top on my todo list.
Feel free to do so if you are motivated, otherwise I'll get to it when I can.

**Pull request merged**